src/Vocup/MainForm.cs

```diff
@@ -125,9 +125,45 @@ public void UnloadBook(bool fullUnload)
             }
         }
 
+        private void StoreSettings()
+        {
+            //
+            // save the forms location and size
+            //
+            Settings.Default.MainFormLocation = this.Location;
+
+            if (this.WindowState == FormWindowState.Normal)
+            {
+                Settings.Default.MainFormSize = this.Size;
+            }
+            else
+            {
+                Settings.Default.MainFormSize = this.RestoreBounds.Size;
+            }
+            Settings.Default.Save();
+        }
+
+        private void RestoreSettings()
+        {
+            //
+            // Restore the MainForm Location and Size
+            //
+            if (Settings.Default.MainFormLocation != null)
+            {
+                this.Location = Settings.Default.MainFormLocation;
+            }
+
+            if (Settings.Default.MainFormSize != null)
```

Comment:
`System.Drawing.Size` is struct which means it will never be `null`. Because the window can't have a size of zero we can treat a size of zero as an uninitialized setting.

```suggestion
            if (Settings.Default.MainFormSize != default)
```